### ISSUE TITLE
Unable to run headless - XServer fails every time

### ISSUE THREAD

**Issue by ch9hn** [games-on-whales/gow#98]
Hello,
Iam unable to run the container with the default settings. So I tried out multiple settings which also failed.

OS: Ubuntu 20.04.5 LTS focal
GPU: NVIDIA RTX3070, Driver Version: 470.141.10   CUDA Version: 11.4

Log:
```
gow-xorg-1      | Detected outputs:
gow-xorg-1      |     - HDMI-0
gow-xorg-1      |     - DP-0
gow-xorg-1      |     - DP-1
gow-xorg-1      |     - DP-2
gow-xorg-1      |     - DP-3
gow-xorg-1      |     - DP-4
gow-xorg-1      |     - DP-5
gow-xorg-1      | 
gow-xorg-1      | Setting DP-0 output to: 1920x1080@60
gow-xorg-1      | 1920x1080 is not detected, FORCE_RESOLUTION=true
gow-xorg-1      | xrandr: cannot find mode 1920x1080
gow-xorg-1      | /opt/gow/startup.sh: line 69: cvt: command not found
gow-xorg-1      | xrandr: failed to parse '1920x1080_60' as a mode specification
gow-xorg-1      | Try 'xrandr --help' for more information.
```

```
$ cat env/headless.env
XORG_IPC=shareable
XORG_VERBOSE=6

SHARED_IPC=service:xorg

###################
# This section is for running in "headless" mode, without a desktop already
# running on your host
###################
XORG_SOCKET=xorg
XORG_DISPLAY=:77

# Change these to suit your preferences
XORG_RESOLUTION=1920x1080
XORG_REFRESH_RATE=60
XORG_DISPLAY_PORT=DP-0 #HDMI-0 #HDMI-0 #DP-0 #DFP-0
XORG_FORCE_RESOLUTION=true

PULSE_SERVER=unix:/tmp/pulse/pulse-socket
PULSE_SOCKET_HOST=pulse

####################
# Services that use udev in order to get devices
# needs to be on the same network
# This is because they need the udev PF_NETLINK socket
UDEVD_NETWORK=service:udevd

```
```
$ cat /etc/X11/xorg.conf

Section "Screen"
    Identifier     "Screen0"
    Device         "Device0"
    Monitor        "Monitor0"
    DefaultDepth    24
    Option         "AllowEmptyInitialConfiguration" "True"
    Option         "UseDisplayDevice" "DP-0"
    Option         "CustomEDID" "DP-0:/home/ubuntu/edid.txt"
    Option         "ConnectedMonitor" "DP-0"
    SubSection     "Display"
        Depth       24
    EndSubSection
EndSection
```
```
$ cat /home/ubuntu/edid.txt
00 ff ff ff ff ff ff 00 1e 6d f5 56 71 ca 04 00 05 14 01 03 80 35 1e 78 0a ae c5 a2 57 4a 9c 25 12 50 54 21 08 00 b3 00 81 80 81 40 01 01 01 01 01 01 01 01 01 01 1a 36 80 a0 70 38 1f 40 30 20 35 00 13 2b 21 00 00 1a 02 3a 80 18 71 38 2d 40 58 2c 45 00 13 2b 21 00 00 1e 00 00 00 fd 00 38 3d 1e 53 0f 00 0a 20 20 20 20 20 20 00 00 00 fc 00 57 32 34 35 33 0a 20 20 20 20 20 20 20 01 3d 02 03 21 f1 4e 90 04 03 01 14 12 05 1f 10 13 00 00 00 00 23 09 07 07 83 01 00 00 65 03 0c 00 10 00 02 3a 80 18 71 38 2d 40 58 2c 45 00 13 2b 21 00 00 1e 01 1d 80 18 71 1c 16 20 58 2c 25 00 13 2b 21 00 00 9e 01 1d 00 72 51 d0 1e 20 6e 28 55 00 13 2b 21 00 00 1e 8c 0a d0 8a 20 e0 2d 10 10 3e 96 00 13 2b 21 00 00 18 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
```

**Comment by zb140** [comment]
Until just a few minutes ago, I hadn't actually tried this before, although I know at least one or two people on our Discord server have mentioned doing something similar in the past.

At first I wasn't sure I'd be able to help much, given my lack of experience, but it turns out I think I can.  Your EDID data is actually incomplete -- the second block is missing its checksum and is therefore only 255 bytes long. Apparently Xorg's EDID parsing isn't happy about that.  If you add ` 26` to the end (that's `26` with a space before it), it should start working.  Be sure to set `XORG_FORCE_RESOLUTION` back to `false`; you shouldn't need it after fixing the EDID issue, and it apparently doesn't work since it's missing `cvt`.  Thanks for finding that!  I'll file a separate issue to track fixing it.

**Comment by ch9hn** [comment]
Thank you very much for you reply.
For everyone who needs a running configuration to run GOW (and Sunshine) on a headless NVIDIA Ubuntu 20.04 Setup, here it is:

# Use this in a situation where you want a headless Linux box with multiple GPU boards but no desktop environment
# Fake EDID convinces drivers that a monitor is connected
# Use any EDID binary file
# To start a fake X server make a systemd file with something like:
# ExecStart=/usr/bin/tcsh -c 'xinit /opt/set-gpu-fans/setfanspeed.sh -- :0 -once -config /opt/set-gpu-fans/xorg.rob'
# See other gist for fan setting script : https://gist.github.com/RobDeBagel/a960c2b157256c162220e60300529cf0
```
$ cat /etx/X11/xorg.conf
Section "ServerLayout"
  Identifier "Layout0"
  Screen  0 "Screen0"
  InputDevice "Keyboard0" "CoreKeyboard"
  InputDevice "Mouse0" "CorePointer"
EndSection

Section "InputDevice"

    # generated from default
    Identifier     "Keyboard0"
    Driver         "keyboard"
EndSection

Section "InputDevice"

    # generated from default
    Identifier     "Mouse0"
    Driver         "mouse"
    Option         "Protocol" "auto"
    Option         "Device" "/dev/psaux"
    Option         "Emulate3Buttons" "no"
    Option         "ZAxisMapping" "4 5"
EndSection

Section "Monitor"
    Identifier     "Monitor0"
    VendorName     "Unknown"
    ModelName      "Unknown"
    HorizSync       28.0 - 33.0
    VertRefresh     43.0 - 72.0
    Option         "DPMS"
EndSection

Section "Device"
    Identifier     "Device0"
    Driver         "nvidia"
    VendorName     "NVIDIA Corporation"
    BoardName      "NVIDIA GeForce RTX 3070"
    BusID          "PCI:1:0:0"
    Option "ConstrainCursor" "off"
    Option "ConnectedMonitor" "DFP-0"
    Option "CustomEDID" "DFP-0:/opt/edid.bin"
EndSection

Section "Screen"
    Identifier     "Screen0"
    Device         "Device0"
    Monitor        "Monitor0"
    DefaultDepth    24
    Option         "UseDisplayDevice" "DFP-0"
    Option         "ConnectedMonitor" "DFP-0"
    Option "AllowEmptyInitialConfiguration" "True"
    SubSection     "Display"
      Depth       24
    EndSubSection
EndSection
```
```
$ cat /opt/edid.bin
00 ff ff ff ff ff ff 00 1e 6d f5 56 71 ca 04 00 05 14 01 03 80 35 1e 78 0a ae c5 a2 57 4a 9c 25 12 50 54 21 08 00 b3 00 81 80 81 40 01 01 01 01 01 01 01 01 01 01 1a 36 80 a0 70 38 1f 40 30 20 35 00 13 2b 21 00 00 1a 02 3a 80 18 71 38 2d 40 58 2c 45 00 13 2b 21 00 00 1e 00 00 00 fd 00 38 3d 1e 53 0f 00 0a 20 20 20 20 20 20 00 00 00 fc 00 57 32 34 35 33 0a 20 20 20 20 20 20 20 01 3d 02 03 21 f1 4e 90 04 03 01 14 12 05 1f 10 13 00 00 00 00 23 09 07 07 83 01 00 00 65 03 0c 00 10 00 02 3a 80 18 71 38 2d 40 58 2c 45 00 13 2b 21 00 00 1e 01 1d 80 18 71 1c 16 20 58 2c 25 00 13 2b 21 00 00 9e 01 1d 00 72 51 d0 1e 20 6e 28 55 00 13 2b 21 00 00 1e 8c 0a d0 8a 20 e0 2d 10 10 3e 96 00 13 2b 21 00 00 18 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 26
```

When you edit the config files, dont forget to reboot.
You can check if the server is running propper with the following xinit line:

`sudo /usr/bin/tcsh -c 'xinit  -- :99 -once'`

**Comment by zb140** [comment]
I'm going to go ahead and close this because the bulk of the issue is resolved, but I've also opened #105 to add some new documentation about how to get this set up for anyone else who needs it in the future.
